File: python/zmq/zguide/hwclient.py
# Hello World client in Python
# Connects REQ socket to tcp://localhost:5555
# Sends "Hello" to server, expects "World" back

import zmq

context = zmq.Context()

print('connecting to hello world server ...')
socket = context.socket(zmq.REQ)
socket.connect('tcp://localhost:5555')

# do 10 requests, waiting each time for a response
for req in range(10):
	print('sending request %s ...' % req)
	socket.send(b'Hello')
	# get replay
	msg = socket.recv()
	print('received reply %s [ %s ]' % (req, msg))

File: zmq3/cppzmq/hwclient.cpp
#include <zmq.hpp>
#include <string>
#include <iostream>

using std::cout;

int main(int argc, char * argv[])
{
	zmq::context_t ctx{1};
	zmq::socket_t socket{ctx, ZMQ_REQ};
	socket.connect("tcp://localhost:5555");

	// do 10 requests/reply
	for (int i = 0; i < 10; ++i)
	{
		zmq::message_t req{5};
		memcpy(req.data(), "Hello", 5);
		cout << "Sending Hello " << i << "..." << std::endl;
		socket.send(req);

		zmq::message_t reply;
		socket.recv(&reply);
		cout << "Received World " << i << std::endl;
	}

	return 0;
}
